go/service/main.go

```diff
@@ -109,11 +110,19 @@ func (d *Service) Handle(c net.Conn) {
 		defer logReg.UnregisterLogger()
 	}
 	shutdowners, err := d.RegisterProtocols(server, xp, connID, logReg, d.G())
-	defer func() {
-		for _, shutdowner := range shutdowners {
-			shutdowner.Shutdown()
-		}
-	}()
+
+	once := &sync.Once{}
```

Comment:
Yes it would I think. But `Do` is defined on `*sync.Once` so I normally tend to be reluctant to use a value type to avoid accidentally copying. But I guess this is fine; will update!